File: dataStudio/Streamis-versions/Streamis-0.2.0/streamis-jobmanager/streamis-job-manager/streamis-job-manager-base/src/main/java/com/webank/wedatasphere/streamis/jobmanager/manager/util/DateUtils.scala
/*
 * Copyright 2021 WeBank
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.webank.wedatasphere.streamis.jobmanager.manager.util

import org.slf4j.LoggerFactory

import java.text.SimpleDateFormat
import java.util.Date

object DateUtils extends Serializable {

  private val logger = LoggerFactory.getLogger(DateUtils.getClass)
  private val serialVersionUID = 1L
  /**
   * Default short date format(默认短日期格式)
   * yyyy-MM-dd
   */
  val DATE_DEFAULT_FORMAT = "yyyy-MM-dd"

  /**
   * Order-number prefix(订单号前缀) yyyyMMddHHmmss
   */
  val DATETIME_ORDER_FORMAT = "yyyyMMddHHmmss"
  /**
   * Default datetime format(默认日期时间格式)
   * yyyy-MM-dd HH:mm:ss
   */
  val DATETIME_DEFAULT_FORMAT = "yyyy-MM-dd HH:mm:ss"
  /**
   * Default time format(默认时间格式)
   * HH:mm:ss
   */
  val TIME_DEFAULT_FORMAT = "HH:mm:ss"
  /**
   * Default date short format(默认日期短格式)
   * yyyyMMdd
   */
  val DATE_DEFAULT_SHORT_FORMAT = "yyyyMMdd"
  /**
   * Default Datetime format(默认日期时间格式化)
   */
  val dateTimeFormat = new SimpleDateFormat(DATETIME_DEFAULT_FORMAT)

  def intervals(start:Date,end:Date): String ={
    if(start == null || end == null) return ""
    val nm = 1000 * 60
    val diff = end.getTime - start.getTime
    (diff/nm)+"分钟"
  }

  def formatDate(dateTime:Date):String = {
    if(dateTime == null) return ""
    dateTimeFormat.format(dateTime)
  }

}

File: dataStudio/Streamis-versions/Streamis-0.2.0/streamis-jobmanager/streamis-projectmanager-server/src/main/java/com/webank/wedatasphere/streamis/projectmanager/utils/DateUtils.java
/*
 * Copyright 2021 WeBank
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.webank.wedatasphere.streamis.projectmanager.utils;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.text.SimpleDateFormat;
import java.util.Date;

public class DateUtils {

    private static final Logger LOG = LoggerFactory.getLogger(DateUtils.class);

    private static final String FORMAT_HH_T_Z = "yyyy-MM-dd'T'HH:mm:ss'Z'";
    private static final String FORMAT_HH_MM_SS = "yyyy-MM-dd HH:mm:ss";
    private static final String FORMAT_HH_MM_SS_S = "yyyy-MM-dd HH:mm:ss.S";
    private static final String FORMAT_HH_MM = "yyyy-MM-dd HH:mm";


    /**
     * contain T,Z format date time convert
     *
     * @param dateTime
     * @return
     * @throws Exception
     */
    public static String dateTimeTZConvert(String dateTime) throws Exception {
        Date date = new SimpleDateFormat(FORMAT_HH_T_Z).parse(dateTime);
        String time = new SimpleDateFormat(FORMAT_HH_MM_SS).format(date);
        return time;
    }

    /**
     * Time format to be reserved to minutes.
     * yyyy-MM-dd HH:mm:ss.S to yyyy-MM-dd HH:mm
     *
     * @param dateTime the amount of time needed to process
     * @return string date time
     */
    public static String timeFormatReservedMinutes(String dateTime) {
        String timeMin = "";
        try {
            SimpleDateFormat sdfSec = new SimpleDateFormat(FORMAT_HH_MM_SS_S);
            Date dt = sdfSec.parse(dateTime);
            SimpleDateFormat sdfMin = new SimpleDateFormat(FORMAT_HH_MM_SS);
            timeMin = sdfMin.format(dt);
        } catch (Exception e) {
            LOG.error("timeFormatReservedMinutes parse error:{}", e);
        }
        return timeMin;
    }

    /**
     * Time format to be reserved to second.
     * yyyy-MM-dd HH:mm:ss.S to yyyy-MM-dd HH:mm:ss
     *
     * @param dateTime the amount of time needed to process
     * @return string date time
     */
    public static String timeFormatReservedSecond(String dateTime) {
        String timeSec = "";
        try {
            SimpleDateFormat sdfSec = new SimpleDateFormat(FORMAT_HH_MM_SS_S);
            Date dt = sdfSec.parse(dateTime);
            SimpleDateFormat sdfMin = new SimpleDateFormat(FORMAT_HH_MM_SS);
            timeSec = sdfMin.format(dt);
        } catch (Exception e) {
            LOG.error("timeFormatReservedMinutes parse error:{}", e);
        }
        return timeSec;
    }
}
